File: src/ProjectR.Scripting/content/scripts/spells/TimeBubble.lua
function GetTargetType()
    return Myself
end

function GetName()
return "Time Bubble"
end

function GetDescription()
    return "Create a time bubble around you that allows you to act faster.\nDoubles your SPD."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 1
end

function GetDelay()
  return .8
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  d:BuffStat(SPD, 1)
end

File: src/ProjectR.Scripting/content/scripts/spells/TimeBubble.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class TimeBubble : SpellScriptBase
    {
        public override string Name { get { return "Time Bubble"; } }
        public override string Description { get { return "Create a time bubble around you that allows you to act faster.\nDoubles your SPD."; } }

        public override TargetType TargetType { get { return TargetType.Myself; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 100; } }
        public override double Delay { get { return .8; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            BuffStat(Stat.SPD, 1);
        }
    }
}